### PR TITLE
cmd/snap-repair: check signatures of repairs from Next

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
based on #3571

**Comment by pedronis** [comment]
remembered it needs a small tweak

**Comment by pedronis** [comment]
addressed detail, reopening

File: cmd/snap-repair/runner.go [hunk 1]
```diff
@@ -625,7 +625,10 @@ func (run *Runner) makeReady(brandID string, sequenceNext int) (repair *asserts.
 			return nil, errSkip
 		}
 	}
-	// TODO: verify with signatures
+	// verify with signatures
+	if err := run.Verify(repair, aux); err != nil {
+		return nil, fmt.Errorf("cannot verify repair %q-%d: %v", brandID, state.Sequence, err)
```
Comment (niemeyer) [inline cmd/snap-repair/runner.go]: s/%q/%s/

File: cmd/snap-repair/runner.go [hunk 2]
```diff
@@ -676,3 +679,35 @@ func (run *Runner) Next(brandID string) (*Repair, error) {
 		}, nil
 	}
 }
+
+// we use designated subroot keys so that we have a subtree of keys and make
```
Comment (niemeyer) [inline cmd/snap-repair/runner.go]: I found it a bit hard to read this sentence.
Perhaps:

_// Limit trust to specific keys while there's no delegation support.
// The obtained assertion stream may also include account keys
// that are directly or indirectly signed by a trusted key._

File: cmd/snap-repair/runner.go [hunk 3]
```diff
@@ -676,3 +679,35 @@ func (run *Runner) Next(brandID string) (*Repair, error) {
...
+// sure not any trusted authority key can sign repairs, given that we lack
+// ATM a way to limit the purpose of keys
+var (
+	trustedRepairRootKeys []*asserts.AccountKey
+)
```
Comment (niemeyer) [inline cmd/snap-repair/runner.go]: We can drop the () while it's a single item, unless you already have follow ups that depend on this.

File: cmd/snap-repair/runner.go [hunk 4]
```diff
@@ -676,3 +679,35 @@ func (run *Runner) Next(brandID string) (*Repair, error) {
...
+
+// Verify verifies that the repair is properly signed taking into consideration account-key assertions from aux as needed.
```
Comment (niemeyer) [inline cmd/snap-repair/runner.go]: We can also use the comment above here, tweaking as needed.

File: cmd/snap-repair/runner.go [hunk 5]
```diff
@@ -676,3 +679,38 @@ func (run *Runner) Next(brandID string) (*Repair, error) {
 		}, nil
 	}
 }
+
+// Limit trust to specific keys while there's no delegation or limited
```
Comment (mvo5) [inline cmd/snap-repair/runner.go]: Thanks for this code comment!

**Pull request merged**